### PR TITLE
Issue #1834 - Move all generated JS files to js/dist

### PR DESCRIPTION
**Pull request by Regaddi**
I think I caught all references, but don't know how to test locally because of `{%- if config.PRODUCTION or config.STAGING -%}`.
Maybe someone has delightful input for me how to test PRODUCTION or STAGING locally 😄 

r? @zoepage

### CODE REVIEW

**Comment by zoepage** [comment]
R? @miketaylr (to be sure we caught everything)

**Comment by miketaylr** [comment]
> Travis tests have failed

(bummer, if https://github.com/webcompat/webcompat.com/pull/2385 was merged, we would have more info on this failure)

**Comment by miketaylr** [comment]
> Maybe someone has delightful input for me how to test PRODUCTION or STAGING locally 😄

@Regaddi you can just set a local environment variable: https://github.com/webcompat/webcompat.com/blob/master/config/environment.py#L19

File: grunt-tasks/concat.js [hunk 1]
```diff
@@ -25,15 +25,15 @@ module.exports = function(grunt) {
         "<%= jsPath %>/lib/bugform.js",
         "<%= jsPath %>/templates.js"
```
Comment (miketaylr) [inline grunt-tasks/concat.js]: Do we need to update this as well?

https://github.com/webcompat/webcompat.com/pull/2394/files#diff-56763bbaaf4762aa0cf49808e18d196eR26

File: .gitignore [hunk 2]
```diff
@@ -20,15 +20,12 @@ data/*
...
-webcompat/static/js/build/
-webcompat/static/js/webcompat.js
-webcompat/static/js/diagnose.js
-webcompat/static/js/issues.js
-webcompat/static/js/issue-list.js
-webcompat/static/js/user-activity.js
-webcompat/static/js/templates.js
 webcompat/**/*.min.js
-webcompat/**/**/*.min.css
+webcompat/static/js/*.js
```
Comment (miketaylr) [inline .gitignore]: Do we need this line? In theory there shouldn't be anything there, right?

(or is it just to ignore the already existing files so people don't have to manually delete them?)

File: grunt-tasks/concat.js [hunk 3]
```diff
@@ -25,15 +25,15 @@ module.exports = function(grunt) {
...
       ],
-      dest: "<%= jsPath %>/<%= pkg.name %>.js"
+      dest: "<%= jsPath %>/dist/<%= pkg.name %>.js"
```
Comment (miketaylr) [inline grunt-tasks/concat.js]: Maybe we could make a new Gruntfile variable`jsDistPath` in Gruntfile.js?

**Comment by zoepage** [comment]
@Regaddi I've merged #2385. Could you rebase please? Thank you! :)

File: .gitignore (hunk 2 above)
Comment (Regaddi) [reply .gitignore]: Yes, it's to make sure, that there are no accidental files or leftovers from older project states.

File: grunt-tasks/concat.js (hunk 3 above)
Comment (Regaddi) [reply grunt-tasks/concat.js]: I can do that, sure.

File: grunt-tasks/concat.js (hunk 1 above)
Comment (Regaddi) [reply grunt-tasks/concat.js]: Since `templates.js` is also a generated file, I'd put it into the `dist` folder as well.

Comment (Regaddi) [reply grunt-tasks/concat.js]: Ah I didn't see it right. Needs to be updated, sure. This made the tests fail I guess.

**Comment by miketaylr** [comment]
Tests are green, LGTM!

**Pull request merged**